client/src/components/LoginPage.js

```diff
@@ -0,0 +1,131 @@
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { DataTable, Button, Panel, TextField, Dialog } from 'lucid-ui';
+
+
+const style = {
+  marginBottom: '10px',
+};
+
+const property = {
+  hasFixedHeader: true,
+  isSelectable: false,
+  hasLightHeader: false,
+  fixedColumnCount: 1,
+  fixedRowHeight: 40,
+  isActionable: false,
+}
+
+class LoginPage extends Component{
+  constructor(props){
+    super(props);
+    this.state = {
+      username: '',
+      createdUsername: '',
+      listUsername: [],
+      status:'',
+      errorMessage:'',
+    }
+  }
+  
+  componentDidMount(){
+    this.getUsernames();
+  }
+
+  clearErrorMessage = () => {
+    this.setState({ errorMessage: '' });
+  }
+
+  getUsernames(){
+    fetch('/api/names/users', {
+      method: 'GET',
+      headers: {
+      'Content-Type': 'application/json',
+      },
+    }).then((response) => {
+      if (!response.ok) {
+          this.setErrorMessage(response.json().content);
```

Comment:
This is weird, whenever you go into a async funciton, the this word changes. so i always create a `const self= this` as a global varible to set state before going into any fetches. but this seems to work on the client.

Reply:
I think using `=>` binds `this` and that's why it works.